auxiliary/eus-multiple-values.l

```diff
@@ -70,15 +73,17 @@
 ;; 	   ,value-form))))
 
 (defmacro multiple-value-bind (varlist value-form &rest body)
+  "Sets each variable in varlist to respective value returned by value-form and subsequently executes body"
   `(multiple-value-call #'(lambda (,@varlist)
-                               ,@body)
-                           ,value-form))
+                            ,@body)
+     ,value-form))
 
 ;; (defmacro multiple-value-setq (varlist value-form)
 ;;   (if varlist
 ;;       `(values (setf (values ,@varlist) ,value-form))
 ;;       `(values ,value-form)))
 
 (defmacro multiple-value-prog1 (form &rest forms)
-  `(progn ,@forms ,form))
-     
+  "Executes form and forms, yielding the value of form"
+  (setq mv-interface::is-mv-on t)
+  `(prog1 ,form ,@forms (setq mv-interface::is-mv-on nil)))
```

Comment:
Probably fine with just `(prog1 ,form ,@forms)`.
Current outcome ↓
```lisp
(setq a (list 1 2 3))
(lisp::multiple-value-prog1 (lisp::values-list a) (setq a (mapcar #'1+ a)))
;; ((mv-interface::multiple . values) 1 2 3)
```

Reply:
The thing is that like this `(prog1 ,form ,@forms)`, multiple-value-prog1 becomes just an alias to prog1, which returns only one value.  I thought it might be better having it return multiple values and not enforcing the rest of the apparatus (if the programmer chose to use `multiple-value-prog` rather than prog1, it seems reasonable to suppose he or she is planning to use multiple values and, as such, would probably use the rest of the apparatus anyway) than just having it return at most one value. But I could be wrong, of course, and would have no problem rolling back to the previous version.

Reply:
Seems to work for multiple-values apparatus with `prog1` as well:
```lisp
(lisp::multiple-value-list (prog1 (values 1 2 3)))
;; (1 2 3)
(1+ (prog1 (values 1 2 3)))
;; 2
```

Reply:
Well, it does work, but it shouldn't, actually. It is because (unlike progn, for instance), [prog1 is defined](http://www.lispworks.com/documentation/HyperSpec/Body/m_prog1c.htm#prog1) to yield only one value (in CLtL we read "prog1 always returns a single value, even if the first form tries to return multiple values."). It might not be a good idea to trust in that it does it right now, because it could make life more difficult if we (or someone else) would want to make things more standard in the future (not that it is a great difference, but we would be making a choice which could haunt us in the future, and it is a choice I'm not sure we really need to make right now).

Reply:
I see. Maybe leave the `multiple-value-prog1` undefined for now and put a note explaining this?

Reply:
ok

Reply:
Or, maybe, we could leave it defined with `progn`. It might sound strange, but that wouldn't get us incompatibility issues in the future (hopefully not, that is) and would make it work as a usual multiple-values form (return multiple values if used with the apparatus, but not otherwise).